platform/android/src/jni.cpp

```diff
@@ -1227,14 +1228,19 @@ void JNICALL nativeAddAnnotationIcon(JNIEnv *env, jobject obj, jlong nativeMapVi
 
     jbyte* pixelData = env->GetByteArrayElements(jpixels, nullptr);
     jsize size = env->GetArrayLength(jpixels);
-    std::string pixels(reinterpret_cast<char*>(pixelData), size);
     env->ReleaseByteArrayElements(jpixels, pixelData, JNI_ABORT);
```

Comment:
This line needs to be moved to after the `std::copy`. (Actually I'm reading now that `GetByteArrayRegion` is a better API to use: http://developer.android.com/training/articles/perf-jni.html.)